config/v1/types_image_digest_mirror_set.go

```diff
@@ -0,0 +1,113 @@
+	// To pull image from mirrors using tag specification, should configure
+	// a list of mirrors using "ImageTagMirrorSet" CRD.
+	//
+	// If the image pull specification matches the reposirtory of "source" in multiple policies,
+	// only the policies have the most specific namespace matching will be used.
+	// For example, if there are policies using quay.io/libpod and quay.io/libpod/busybox as
+	// the "source", only the policies using quay.io/libpod/busybox are going to apply
+	// for pull specification quay.io/libpod/busybox.
+	// Each “source” repository is treated independently; configurations for different “source”
+	// repositories don’t interact.
+	//
+	// If the "mirrors" is not specified, the image will continue to be pulled from the specified
+	// repository in the pull spec.
+	//
+	// When multiple policies are defined for the same “source” repository, the sets of defined
+	// mirrors will be merged together, preserving the relative order of the mirrors, if possible.
+	// For example, if policy A has mirrors `a, b, c` and policy B has mirrors `c, d, e`, the
+	// mirrors will be used in the order `a, b, c, d, e`.  If the orders of mirror entries conflict
+	// (e.g. `a, b` vs. `b, a`) the configuration is not rejected but the resulting order is unspecified.
+	// Users who want to use a specific order of mirrors, should configure them into one list of mirrors using the expected order.
+	// +optional
+	// +listType=atomic
+	ImageDigestMirrors []ImageDigestMirrors `json:"imageDigestMirrors"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageDigestMirrorSetList lists the items in the ImageDigestMirrorSet CRD.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type ImageDigestMirrorSetList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []ImageDigestMirrorSet `json:"items"`
+}
+
+// ImageDigestMirrors holds cluster-wide information about how to handle mirrors in the registries config.
+type ImageDigestMirrors struct {
+	// source matches the repository that users refer to, e.g. in image pull specifications.
+	// "source" uses one of the following formats:
+	// host[:port]
+	// host[:port]/namespace[/namespace…]
+	// host[:port]/namespace[/namespace…]/repo
+	// [*.]host
+	// for more information about the format, see the document about the location field:
+	// https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md#choosing-a-registry-toml-table
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^\*(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+$|^((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?)(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?$`
```

Comment:
@bparees deserves to review this regex because likes them :)

Reply:
FYI, used helper from containers/image https://github.com/containers/image/blob/main/docker/reference/regexp.go to make these regex: https://go.dev/play/p/NO9_2LmqPiu

Reply:
i figured(hoped) you'd pulled them from somewhere :)